File: src/main/java/app/controller/data/RouteTypeController.java
package app.controller.data;

import app.dto.models.RouteTypeDto;
import app.service.entities.IRouteTypeService;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;

import javax.validation.Valid;
import java.util.List;

@Controller
@RequestMapping(value = "/routeType")
public class RouteTypeController {
    static final Logger logger = LoggerFactory.getLogger(RouteTypeController.class);

    @Autowired
    IRouteTypeService service;

    //CREATE ONE ROUTETYPE
    @RequestMapping(value = "/", method = RequestMethod.POST,  produces = MediaType.APPLICATION_JSON_VALUE)
    public ResponseEntity<Void> create(@Valid RouteTypeDto entityDto) {
        logger.info("create {}", entityDto);
        boolean isCreated = service.create(entityDto);
        if (!isCreated) {
            return new ResponseEntity<>(HttpStatus.CONFLICT);
        }
        return new ResponseEntity<>(HttpStatus.CREATED);
    }

    //GET ROUTE
    @RequestMapping(value = "/{marker}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
    public ResponseEntity<RouteTypeDto> read (@PathVariable("marker") String marker) {
        RouteTypeDto currentDto = service.read(marker);
        logger.info("found entity: ", currentDto);
        if (currentDto == null) {
            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
        }
        return new ResponseEntity<>(currentDto, HttpStatus.OK);
    }

    //CREATE OR UPDATE ROUTETYPE
    @RequestMapping(method = RequestMethod.PUT, produces = MediaType.APPLICATION_JSON_VALUE)
    public ResponseEntity<Void> updateOrUpdate(@RequestBody @Valid RouteTypeDto entityDto) {
        logger.info("createOrUpdate: {}", entityDto);
        if (service.createOrUpdate(entityDto)) {
            return new ResponseEntity<>(HttpStatus.OK);
        } else {
            return new ResponseEntity<>(HttpStatus.CONFLICT);
        }
    }

    //UPDATE ROUTETYPE
    @RequestMapping(value = "/{marker}", method = RequestMethod.PUT, produces = MediaType.APPLICATION_JSON_VALUE)
    public ResponseEntity<Void> update(@PathVariable("marker") String marker, @RequestBody @Valid RouteTypeDto entityDto) {
        logger.info("update by marker: {}", marker);
        if (service.update(marker, entityDto)) {
            return new ResponseEntity<>(HttpStatus.OK);
        } else {
            return new ResponseEntity<>(HttpStatus.CONFLICT);
        }
    }

    //DELETE ROUTETYPE
    @RequestMapping(value = "/{marker}", method = RequestMethod.DELETE, produces = MediaType.APPLICATION_JSON_VALUE)
    public ResponseEntity<Void> delete(@PathVariable("marker") String marker) {
        logger.info("delete: {}", marker);
        if (service.delete(marker)) {
            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
        } else {
            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
        }
    }

    //GET ALL ROUTETYPES
    @RequestMapping(method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
    public ResponseEntity<List<RouteTypeDto>> listAll() {
        logger.info("get all items");
        List<RouteTypeDto> entityDtoList = service.getAll();
        if(entityDtoList.isEmpty()){
            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
        }
        return new ResponseEntity<>(entityDtoList, HttpStatus.OK);
    }

}

File: src/main/webapp/resources/js/data/routeTypeController.js
app.controller('routeTypeController', function ($scope, entityService) {
    $scope.routeTypes = [];

    entityService.get('/routeType', function (data) {
        console.log('get all items works...');
        $scope.routeTypes = data;
    });

    $scope.newRouteType = {
        marker: undefined,
        name: undefined
    };
    $scope.newRouteTypes = [];

    $scope.getAndShow = function () {
        console.log('getAndShow works...' + $scope.newRouteType.marker);
        entityService.get('/routeType/' + $scope.newRouteType.marker, function (data) {
            console.log(data);
            if (data) {
                $scope.newRouteType = data;
            }
        })
    };

    $scope.addNewToTemp = function () {
        console.log('addNewItem to temp collection works...');
        var entity = angular.copy($scope.newRouteType);
        $scope.newRouteTypes.push(entity);
        console.log(entity);
    };

    $scope.sendNewEntriesToDatabase = function () {
        console.log('sendNewEntriesToDatabase works...');
        entityService.createOrUpdate('/routeType', $scope.newRouteTypes)
    };

    $scope.deleteFromDB = function (marker) {
        console.log('deleteFromDB  works...' + marker);
        entityService.delete('/routeType/' + marker)
    };
});